src/app/shared/components/template/services/instance/template-action.service.ts

```diff
@@ -18,17 +24,29 @@ let log_groupEnd = SHOW_DEBUG_LOGS ? console.groupEnd : () => null;
  *
  *
  */
-export class TemplateActionService {
+export class TemplateActionService extends TemplateInstanceService {
   private actionsQueue: FlowTypes.TemplateRowAction[] = [];
   private actionsQueueProcessing$ = new BehaviorSubject<boolean>(false);
 
-  constructor(
-    public container: TemplateContainerComponent,
-    private settingsService: SettingsService,
-    private serverService: ServerService,
-    private analyticsService: AnalyticsService,
-    private templateFieldService: TemplateFieldService
-  ) {}
+  private settingsService: SettingsService;
+  private serverService: ServerService;
+  private analyticsService: AnalyticsService;
+  private templateNavService: TemplateNavService;
+  private templateService: TemplateService;
+  private tourService: TourService;
+  private templateTranslateService: TemplateTranslateService;
+  private templateFieldService: TemplateFieldService;
+
+  constructor(public container: TemplateContainerComponent, injector: Injector) {
+    super(injector);
+    this.settingsService = this.getGlobalService(SettingsService);
+    this.serverService = this.getGlobalService(ServerService);
+    this.analyticsService = this.getGlobalService(AnalyticsService);
+    this.templateNavService = this.getGlobalService(TemplateNavService);
+    this.templateService = this.getGlobalService(TemplateService);
+    this.tourService = this.getGlobalService(TourService);
+    this.templateFieldService = this.getGlobalService(TemplateFieldService);
+  }
```

Comment:
This is the change to how services are shared to instantiated services (the template-actions service being one)
Instead of passing all services via args (what is in the `constructor` script), it instead defines placeholders which are then populated via a `getGlobalService` method